projects/scion/toolkit.internal/widgets/src/tabbar/tab.directive.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018-2019 Swiss Federal Railways
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ */
+
+import { Directive, Input, OnDestroy, TemplateRef, ViewContainerRef, ViewRef } from '@angular/core';
+
+/**
+ * Use this directive to model a tab item for {SciTabbarComponent}.
+ * The host element of this modelling directive must be a <ng-template>.
+ *
+ * Tab content is constructed lazily when displayed for the first time. Tab content is not destroyed when selecting another tab.
+ *
+ * ---
+ * Example usage:
+ *
+ * <sci-tabbar>
+ *   <ng-template sciTab [label]="contact.name" *ngFor="let contact of contacts$ | async">
+ *     <app-contact-list-item [contact]="contact"></app-contact-list-item>
+ *   </ng-template>
+ * </sci-tabbar>
+ */
+@Directive({selector: 'ng-template[sciTab]'})
+export class SciTabDirective implements OnDestroy {
+
+  private _vcr: ViewContainerRef | undefined;
+  private _viewRef: ViewRef | undefined;
+
+  /**
+   * Specifies the title of the tab.
+   */
+  @Input()
+  public label: string;
+
+  constructor(private readonly _templateRef: TemplateRef<void>) {
+  }
+
+  /**
+   * Attaches the content of this tab.
+   *
+   * @param vcr
+   *        Specifies where to attach this tab's content.
+   */
+  public attachContent(vcr: ViewContainerRef): void {
+    this._vcr = vcr;
+
+    // Construct the view, if not already constructed.
+    if (!this._viewRef) {
+      this._viewRef = this._templateRef.createEmbeddedView(null);
+    }
+
+    // Attach the content, if not already attached.
+    if (!this.isContentAttached()) {
+      this._viewRef.reattach();
+      vcr.insert(this._viewRef);
+    }
+  }
+
+  /**
+   * Detaches the content of this tab, but does not destroy it.
+   */
+  public detachContent(): void {
+    if (this.isContentAttached()) {
+      this._vcr.detach(this._vcr.indexOf(this._viewRef));
+      this._viewRef.detach();
+    }
+  }
+
+  /**
+   * Returns if this tab's content is currently attached, i.e., visible to the user.
+   */
+  public isContentAttached(): boolean {
+    return !!this._vcr && this._vcr?.indexOf(this._viewRef) !== -1;
```

Comment:
If using the safe navigation operator, you can simplify it as following:
```
return this._vcr?.indexOf(this._viewRef) !== -1 || false;
```

Reply:
done